core/ai.go

```diff
@@ -23,14 +24,40 @@ type AI interface {
 	HasCapacity(pipeline, modelID string) bool
 }
 
+// Custom type to handle both string and int but store as string.
+type StringInt string
```

Comment:
@victorges I'd love to hear your thoughts on this. I implemented this change to allow both integers and strings in the AIModel config for backward compatibility. However, this approach removes static typing and is a bit of a go anti-patern. If you think switching entirely to strings in the config file is a better solution, please let me know 👍🏻.

Reply:
Hm, a different approach here would be to just add another field for the `Currency`, like it was done for the per-gateway prices config on the original PR that implemented USD pricing: https://github.com/livepeer/go-livepeer/pull/2981/files#diff-8e2167038b87d117b916131f62552da8dee7f52d8112e490330349597b393617R1473

I think this renders a clearer interface for the JSON, with no multi-type fields, and is also fully retro-compatible. It would also be good to keep a consistent schema between different configs for similar things.

Reply:
@victorges I like the method you suggested in [this commit](https://github.com/livepeer/go-livepeer/pull/3103/commits/8036856cbfd6724f79b393118e49fd2451282797). The only issue is that it no longer works with the old configuration format. If I understand correctly, you'd like users to specify the config in this way instead of:

```json
[
  {
    "pipeline": "text-to-image",
    "model_id": "SG161222/RealVisXL_V4.0_Lightning",
    "price_per_unit": "16USD",
    "warm": true
  }
]
```

changing it to:

```json
[
  {
    "pipeline": "text-to-image",
    "model_id": "SG161222/RealVisXL_V4.0_Lightning",
    "price_per_unit": 16,
    "currency": "USD",
    "warm": true
  }
]
```

Does that capture your intention correctly? I think that approach is clean. The reason I didn't initially go with it was to keep it similar to how the `pricePerUnit` flag worked on the transcoding side. However, since we use JSON files or JSON strings, this makes sense. 👍🏻

Reply:
One side effect of having non-pixel-related pipelines now is that `PixelsPerUnit     JSONRat                  `json:"pixels_per_unit,omitempty"` is no longer representative of its function. It acts more as a scaling factor than a true `pixelsPerUnit`. Perhaps we should rename it to `ScalingFactor`? Tagging @ad-astra-video, @eliteprox, @pschroedl for visibility.

Reply:
@victorges I just see https://github.com/livepeer/go-livepeer/pull/3103#discussion_r1700935708 👍🏻.

Reply:
@victorges I added the currency json field. See https://github.com/livepeer/go-livepeer/pull/3103#discussion_r1701635888 and https://github.com/livepeer/go-livepeer/pull/3103/commits/8e654d7ad60b68b609be069ebcdfd045978102f6.

Reply:
new version is perfect!

Reply:
> One side effect of having non-pixel-related pipelines now is that PixelsPerUnit     JSONRat is no longer representative of its function (see https://vscode.dev/github/livepeer/go-livepeer/blob/ai-video-rebase/core/ai.go#L51).. It acts more as a scaling factor than a true pixelsPerUnit. Perhaps we should rename it to ScalingFactor?

Yeah I think that could make sense. If we rename this I'd say we should go all the way and think of consistent naming to use for both parameters. I don't think "price per unit" would make that much sense if there's no "unit" mentioned anywhere else for example. We could also use terms like "numerator" and "denominator" to be super clear how they are used, it was pretty confusing for operators when they had to start using this `pixelsPerUnit` flag (and I always have to think twice about it).

TLDR Open to renaming but I think we should have a small discussion about what it should be.

And FTR, I don't think it's a big issue to keep it with a weird name for now. Just the conflicts we're going to have by renaming it here might be enough to avoid it.